todo-app/src/App.js

```diff
@@ -1,20 +1,135 @@
 import React, { Component } from 'react';
 import logo from './logo.svg';
 import './App.css';
+import todo from './todo.json';
+import TasksForm  from './components/TasksForm';
+import AddTodo from './components/AddTodo';
+import DeleteAllButton from './components/DeleteAllButton';
+
 
 class App extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      todos: [
+        {
+          description: "Study Math",
+          done: false,
+          id:1
+        },
+        {
+          description: "Finish your homework",
+          done: true,
+          id:2
+        },
+        {
+          description: "Eat your sandwich",
+          done: true,
+          id:3
+        },
+        {
+          description: "Listen to music",
+          done: true,
+          id:4
+        }
+      ],
+      lastGeneratedId:4
+    }
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleDelete = this.handleDelete.bind(this)
+    this.handleCompletedTask = this.handleCompletedTask.bind(this)
+    this.handleNewTaskSubmit = this.handleNewTaskSubmit.bind(this)
+    this.handleDeleteAll = this.handleDeleteAll.bind(this)
+  }
+
+handleSubmit(newDescription,id){
+  const index = this.state.todos.findIndex((item) => {
+    return item.id === id
+  })
+  const todo = this.state.todos[index]
+  const newTodo = Object.assign({}, todo, {
+    description: newDescription
+  })
+  this.setState({
+    todos: this.state.todos.slice(0, index).concat(newTodo, this.state.todos.slice(index + 1))
+  })
+  console.log(index)
+  console.log(newDescription + id)
+}
+
+handleDelete(newId){
```

Comment:
newId is here refering to the ID but I needed to give it a different name for the extinguish.
Splice will return the deleted item, which will leave me with only the item I wanted to delete instead of the array without the item I deleted